mlrun/datastore/sources.py

```diff
@@ -875,21 +873,45 @@ def __init__(
         collection_name: str = None,
         spark_options: dict = None,
     ):
-        key = "MONGO_CONNECTION_STRING"
-        connection_string = connection_string or os.getenv(key)
+        """
+        Reads MongoDB as input source for a flow.
+
+        example::
+             connection_string = "???"
+             query = {age: {"$gt: 5}}
+             MongoDBSource(connection_string=connection_string, collection_name="coll",
+             db_name="my_dataset", chunksize=5, query=query)
+
+        :param name:  source name
+        :param query: dictionary query for mongodb
+        :param chunksize: number of rows per chunk (default large single chunk)
+        :param key_field: the column to be used as the key for events. Can be a list of keys.
+        :param time_field: the column to be parsed as the timestamp for events. Defaults to None
+        :param start_time: filters out data before this time
```

Comment:
why `start_time` and `end_time` do not have type hints? If they can be multiple types, you can use `typing.Union[int, str]` for exmaple to be either `str` or `int`.